File: docs/code-examples/line_segments3d_simple.cpp
// Log a simple set of line segments.

#include <rerun.hpp>

int main() {
    auto rec = rerun::RecordingStream("rerun_example_line_segments3d");
    rec.connect().throw_on_failure();

    std::vector<std::vector<std::array<float, 3>>> points = {
        {{0.f, 0.f, 0.f}, {0.f, 0.f, 1.f}},
        {{1.f, 0.f, 0.f}, {1.f, 0.f, 1.f}},
        {{1.f, 1.f, 0.f}, {1.f, 1.f, 1.f}},
        {{0.f, 1.f, 0.f}, {0.f, 1.f, 1.f}},
    };

    rec.log("segments", rerun::LineStrips3D(points));
}

File: docs/snippets/all/line_segments3d_simple.py
#!/usr/bin/env python3
"""Log a simple set of line segments."""
import numpy as np
import rerun as rr

rr.init("rerun_example_line_segments3d", spawn=True)

rr.log(
    "segments",
    rr.LineStrips3D(
        np.array(
            [
                [[0, 0, 0], [0, 0, 1]],
                [[1, 0, 0], [1, 0, 1]],
                [[1, 1, 0], [1, 1, 1]],
                [[0, 1, 0], [0, 1, 1]],
            ],
        )
    ),
)
